tests/utils/test_utils.py

```diff
@@ -123,7 +123,7 @@ def test_gen_design():
         'gen_design does not treat missing values correctly')
     design5 = gen_design(stimtime_files=[files['FSL2']],
                          scan_duration=[48, 20], TR=1)
-    assert np.all(np.isclose(design4, design5[::2])), (
+    assert (design4 - design5[::2]).mean() < 0.1, (
```

Comment:
Is this still necessary given you have used direct down-sampling?
If it is still necessary. Maybe it should be the absolute value of difference that is smaller than some threshold, and 0.1 on the average difference may be too loose.

Reply:
Up to you to decide whether it is still necessary.

I tested this value and it seemed in an appropriate range for the average difference. Do you have an alternative suggestion?

Reply:
Ah I see, I thought they will become exactly the same so `np.all(np.isclose...)` can actually work. But maybe I am wrong. 
Oh I think I know why, you are using the time of the middle slice. So then design4 and design5 are guaranteed to be different (they are off by 0.5 s). Maybe we should just remove this test between design4 and design5, given your current implementation.

Reply:
Yes that is exactly right, the middle slice makes it different than if it was the first or last slice. I actually like this test though since they should be similar, just not the exact same. I vote to keep it in so that we know future changes to these functions don't have a big effect

Reply:
OK! Let's keep it. Does `assert (np.abs(design4 - design5[::2])).mean() < 0.1` work? I just think the mean should be applied to some metric of distance, rather than just the subtraction. If not, I think decreasing TR would make it work.